crates/cast/src/lib.rs

```diff
@@ -350,6 +350,26 @@ pub fn print_formatted(output: Vec<(&str, String)>, json: bool, error: bool) ->
     Ok(())
 }
 
+fn json_value_to_string(value: &Value, value_format: ValueFormat) -> Option<String> {
+    let res = match value {
+        Value::Number(n) => {
+            let n = n.as_u64().expect("found unexpected value");
+            value_format.format_u64(n)
+        }
+        Value::String(s) => value_format.format_str(s),
+        Value::Array(arr) => {
+            let arr_as_string = arr
+                .iter()
+                .filter_map(|item| json_value_to_string(item, value_format))
+                .collect::<Vec<String>>()
+                .join(", ");
+            format!("[{arr_as_string}]")
+        }
+        _ => return None,
+    };
+    Some(res)
+}
+
```

Comment:
I removed the `res` variable.
As for the other points:

- there are no ownership issues with `value_format` since it implements the `Copy` trait
- `n.as_u64()` we may want this to `panic`, so we know that invalid values are encountered and we have a problem somewhere in our code

Reply:
cool, thanks! as for:

> n.as_u64() we may want this to panic

we definitely want avoid panic-ing, we should rather display informative and concise error messages

Reply:
If we use `map`, it will not print anything
Do you suggest using a different error message in `expect`?

Reply:
yes, expect should be enough here

Reply:
improved the error msg